thorlcr/activities/indexwrite/thindexwrite.cpp

```diff
@@ -346,34 +345,17 @@ class IndexWriteActivityMaster : public CMasterActivity
                 mb.read(crc);
                 props.setPropInt64("@fileCrc", crc);
 
-                unsigned __int64 slaveNumLeafNodes;
-                unsigned __int64 slaveNumBlobNodes;
-                unsigned __int64 slaveNumBranchNodes;
-                offset_t slaveOffsetBranches;
                 offset_t slaveUncompressedSize;
                 offset_t slaveOriginalBlobSize;
-                offset_t slaveBranchMemorySize;
-                offset_t slaveLeafMemorySize;
-                mb.read(slaveNumLeafNodes);
-                mb.read(slaveNumBlobNodes);
-                mb.read(slaveNumBranchNodes);
-                mb.read(slaveOffsetBranches);
                 mb.read(slaveUncompressedSize);
                 mb.read(slaveOriginalBlobSize);
-                mb.read(slaveBranchMemorySize);
-                mb.read(slaveLeafMemorySize);
 
                 compressedFileSize += size;
-                numLeafNodes += slaveNumLeafNodes;
-                numBlobNodes += slaveNumBlobNodes;
-                numBranchNodes += slaveNumBranchNodes;
                 uncompressedSize += slaveUncompressedSize;
                 originalBlobSize += slaveOriginalBlobSize;
-                branchMemorySize += slaveBranchMemorySize;
-                leafMemorySize += slaveLeafMemorySize;
 
                 props.setPropInt64("@uncompressedSize", slaveUncompressedSize);
-                props.setPropInt64("@offsetBranches", slaveOffsetBranches);
+                props.setPropInt64("@offsetBranches", statsCollection.getStatisticSum(StSizeOffsetBranches));
```

Comment:
The call to getStatisticSum() looks wrong.  This needs to be the value for this particular file part, not the sum for all file parts.
If it is wrong this needs fixing before the release goes gold.

Did you run processes on a multi part thor before and afterwards and check that the same numbers were generated?

Reply:
@ghalliday I did execute on multi part Thor.  I think I still have the test workunits.  I'll reexamine the stats.